src/mpid/ch4/src/ch4r_recvq.h

```diff
@@ -158,13 +158,18 @@ MPL_STATIC_INLINE_PREFIX MPIR_Request *MPIDIG_dequeue_posted(int rank, int tag,
     MPIR_T_PVAR_TIMER_START(RECVQ, time_failed_matching_postedq);
     DL_FOREACH_SAFE(*list, curr, tmp) {
         MPIR_T_PVAR_COUNTER_INC(RECVQ, posted_recvq_match_attempts, 1);
-        req = curr->request;
-        if (MPIDIG_match_posted(rank, tag, context_id, req)) {
+#ifndef MPIDI_CH4_DIRECT_NETMOD
+        /* NOTE: extra negation to force logical comparisons */
+        if (!MPIDI_REQUEST(curr->request, is_local) != !is_local) {
```

Comment:
Personally I don't like this. Can we change the logic a little bit to get rid of the extra negation? Would something like:
```
    bool do_matching = true;

#ifndef MPIDI_CH4_DIRECT_NETMOD
    do_matching = (MPIDI_REQUEST(curr->request, is_local) && is_local);
#endif 

    if (do_matching && MPIDIG_match_posted(rank, ...)) {
        ...
        break;
    }
```
work?

Reply:
From our offline discussion, we decided to let this one go as the alternative is even more gross :(